src/pages/Signup.jsx

```diff
@@ -0,0 +1,304 @@
+// React Imports
+import React, { useState, useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
+// Inrupt Imports
+import {
+  createAcl,
+  createContainerAt,
+  getResourceAcl,
+  getSolidDatasetWithAcl,
+  getThing,
+  getWebIdDataset,
+  setAgentDefaultAccess,
+  setAgentResourceAccess,
+  saveAclFor,
+  getStringNoLocale
+} from '@inrupt/solid-client';
+import { FOAF } from '@inrupt/vocab-common-rdf';
+// Material UI imports
+import { TextField, Button, CardHeader } from '@mui/material';
+import Box from '@mui/material/Box';
+import Card from '@mui/material/Card';
+import Container from '@mui/material/Container';
+import Paper from '@mui/material/Paper';
+import Typography from '@mui/material/Typography';
+// Hooks
+import useContactsList from '@hooks/useContactsList';
+import useSession from '@hooks/useSession';
+
+import { ENV } from '@constants';
+
+const formRowStyle = {
+  margin: '20px 0'
+};
+
+const textFieldStyle = {
+  margin: '8px'
+};
+
+const cardStyle = {
+  padding: '8px',
+  margin: '8px',
+  boxShadow: '0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)'
+};
+
+const registerPod = async ({ email, password, confirmPassword }, oidcProvider) => {
+  const [podName] = email.split('@');
+
+  const oidcRegistrationPath = `${oidcProvider}idp/register/`;
+
+  const body = {
+    email,
+    password,
+    confirmPassword,
+    podName,
+    createWebId: true,
+    createPod: true,
+    rootPod: false,
+    register: true
+  };
+
+  const response = await fetch(oidcRegistrationPath, {
```

Comment:
There seems to be an error related to the fetch function in `Signup.jsx`. The URL for pod registration seems to be incorrect when testing it out on localhost `http://localhost:3000idp/register/` it seems to be missing the `/` between the server URL and the path for registration.

Reply:
Nevermind. Think this was more of an .env thing. Got this fixed.